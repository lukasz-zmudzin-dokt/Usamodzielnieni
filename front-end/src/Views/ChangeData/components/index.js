export { default as FacilityForm } from "./FacilityForm";
export { default as CompanyForm } from "./CompanyForm";
