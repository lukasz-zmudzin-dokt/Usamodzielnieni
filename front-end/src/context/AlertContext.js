import React, { useState, createContext } from "react";

export const AlertContext = createContext({
  open: false,
  message: "",
  setOpen: () => {},
});

export const AlertProvider = (props) => {
  const [open, setOpen] = useState(false);
  const [message, setMessage] = useState("");
  const [variant, setVariant] = useState("danger") // success
  const data = {
    open,
    message,
    variant,
    changeVisibility: (newOpen) => setOpen(newOpen),
    changeMessage: (newMessage,newVariant = variant) => {
      setOpen(false);
      setMessage(newMessage);
      setVariant(newVariant);
    },
    showAlert:  (newMessage,newVariant) => {
      data.changeMessage(newMessage,newVariant);
      data.changeVisibility(true);
    }
  };
  return <AlertContext.Provider value={data} {...props} />;
};
