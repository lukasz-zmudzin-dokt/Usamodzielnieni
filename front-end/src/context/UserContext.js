import React, { useState } from "react";
import Cookies from "universal-cookie";

export const UserContext = React.createContext({
  type: undefined, // Employer,Standard,Staff
  token: undefined,
  login: () => {},
  logout: () => {}
});
const cookies = new Cookies();

export const UserProvider = props => {
  const [type, setType] = useState(cookies.get("type"));
  const [token, setToken] = useState(cookies.get("token"));
  const user = {
    token,
    type,
    login: (newToken, newType) => {
      cookies.set("token", newToken, { path: "/" });
      cookies.set("type", newType, { path: "/" });
      setToken(newToken);
      setType(newType);
    },
    logout: () => {
      cookies.remove("token", { path: "/" });
      cookies.remove("type", { path: "/" });
      setToken(undefined);
      setType(undefined);
    }
  };
  return <UserContext.Provider value={user} {...props} />;
};
