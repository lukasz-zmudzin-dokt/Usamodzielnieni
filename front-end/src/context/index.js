export { UserContext, UserProvider } from "context/UserContext";
export { AlertContext, AlertProvider } from "context/AlertContext";
export { ChatContext, ChatProvider } from "context/ChatContext";
export {
  NotificationsContext,
  NotificationsProvider,
} from "context/NotificationsContext";
