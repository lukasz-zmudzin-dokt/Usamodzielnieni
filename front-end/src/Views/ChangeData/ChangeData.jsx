import React, { useState, useEffect, useContext, useRef } from "react";
import { FormGroup } from "components";
import { Form, Card, Button, Container, Row, Alert } from "react-bootstrap";
import { getUserData, sendFixedData } from "./functions/changeData";
import { useHistory, useParams } from "react-router-dom";
import { UserContext, AlertContext } from "context";
import { FacilityForm, CompanyForm } from "./components";

const ChangeData = () => {
  const [data, setData] = useState({
    email: "",
    first_name: "",
    last_name: "",
    phone_number: "",
    facility_name: "",
    facility_address: {
      city: "",
      street: "",
      street_number: "",
      postal_code: "",
    },
  });
  const [validated, setValidated] = useState(false);
  const [loading, setLoading] = useState(false);
  const [disabled, setDisabled] = useState(true);
  const [err, setErr] = useState(false);
  const history = useHistory();
  const { id } = useParams();
  const user = useContext(UserContext);
  const alertC = useRef(useContext(AlertContext));

  const backToList = () => {
    history.push("/userList");
  };

  //standard b582b042-d6d8-4e57-9447-564a6748b4f7
  //employer fe2c0fca-91a1-437d-846e-7fa6fe6e87ed

  useEffect(() => {
    setLoading(true);
    setDisabled(true);
    const getData = async (token, id) => {
      try {
        const res = await getUserData(token, id);
        setData(res);
      } catch (err) {
        setErr(true);
      }
      setLoading(false);
      setDisabled(false);
    };
    getData(user.token, id);
  }, [id, user.token]);

  const { first_name, last_name, phone_number, email } = data;

  const handleSubmit = async (e) => {
    e.preventDefault();
    const form = e.currentTarget;

    if (form.checkValidity() !== false) {
      setDisabled(true);
      try {
        await sendFixedData(user.token, id, data);
        alertC.current.showAlert(
          "Udało się przesłać poprawione dane.",
          "success"
        );
        return history.push("/userList");
      } catch (err) {
        alertC.current.showAlert("Nie udało się przesłać danych.");
      }
    }
    setValidated(true);
    setDisabled(false);
  };

  return (
    <Container>
      <Card>
        <Card.Header as="h2" className="changeData__header">
          <Button
            className="changeData__back"
            onClick={backToList}
            variant="outline-warning"
          >
            {"<"}
          </Button>
          Popraw dane użytkownika:{" "}
        </Card.Header>
        <Card.Body>
          {err ? (
            <Alert variant="danger">
              Nie udało się pobrać danych użytkownika.
            </Alert>
          ) : (
            <Form
              className="changeData__form"
              noValidate
              validated={validated}
              onSubmit={(e) => handleSubmit(e)}
            >
              <div className="changeData__wrapper">
                <Card bg="light" className="changeData__wrapper__card">
                  <Card.Header>Dane osobowe</Card.Header>
                  <Card.Body>
                    <FormGroup
                      header="Imię"
                      setVal={(val) => setData({ ...data, first_name: val })}
                      val={first_name}
                      length={{ min: 1, max: 30 }}
                      required
                      id="firstName"
                      incorrect="Podaj imię użytkownika."
                    />
                    <FormGroup
                      header="Nazwisko"
                      setVal={(val) => setData({ ...data, last_name: val })}
                      val={last_name}
                      length={{ min: 1, max: 30 }}
                      required
                      id="lastName"
                      incorrect="Podaj nazwisko użytkownika."
                    />
                    {data.group_type ? null : (
                      <FormGroup
                        header="Numer telefonu"
                        type="tel"
                        setVal={(val) =>
                          setData({ ...data, phone_number: val })
                        }
                        val={phone_number}
                        pattern="[+]{1}[4]{1}[8]{1}[0-9]{3}[0-9]{3}[0-9]{3}"
                        id="phoneNumber"
                        incorrect="Podaj numer telefonu użytkownika w formacie +48123123123."
                        required
                      />
                    )}
                  </Card.Body>
                </Card>
                <Card bg="light" className="changeData__wrapper__card mb-3">
                  <Card.Header>Dane konta</Card.Header>
                  <Card.Body>
                    <FormGroup
                      header="Email"
                      type="email"
                      setVal={(val) => setData({ ...data, email: val })}
                      val={email}
                      length={{ min: 1, max: 60 }}
                      required
                      id="email"
                      incorrect="Podaj poprawny email użytkownika."
                    />
                  </Card.Body>
                </Card>
                {loading ? (
                  <Alert variant="info" className="changeData__loading">
                    Ładowanie...
                  </Alert>
                ) : data.nip ? (
                  <CompanyForm data={data} setData={setData} />
                ) : data.group_type ? null : (
                  <FacilityForm data={data} setData={setData} />
                )}
              </div>
              <Row className="ml-0 mr-0 mt-3 justify-content-center">
                <Button type="submit" disabled={disabled} variant="primary">
                  {disabled ? "Ładowanie... " : "Prześlij zmiany"}
                </Button>
              </Row>
            </Form>
          )}
        </Card.Body>
      </Card>
    </Container>
  );
};

export default ChangeData;
