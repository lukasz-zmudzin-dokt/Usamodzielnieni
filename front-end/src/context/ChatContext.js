import React, {
  createContext,
  useEffect,
  useState,
  useContext,
  useCallback,
} from "react";
import proxy from "config/api";
import { UserContext } from "context";

const getChats = async (token, filters) => {
  let url = `${proxy.chat}?page=${filters.page}&page_size=${filters.pageSize}`;
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };

  const response = await fetch(url, { method: "GET", headers });

  if (response.status === 200) {
    return response.json().then((chats) => chats);
  } else {
    throw response.status;
  }
};

export const ChatContext = createContext(null);

export const ChatProvider = (props) => {
  const user = useContext(UserContext);
  const [chats, setChats] = useState([]);
  const [isChatsLoading, setIsChatsLoading] = useState(false);
  const [error, setError] = useState(false);
  const [count, setCount] = useState(0);
  const [filters] = useState({
    page: 1,
    pageSize: 10,
  });
  const [activePage, setActivePage] = useState(1);

  useEffect(() => {
    if (user.token) {
      const loadChats = async (token) => {
        setActivePage(1);
        setError(false);
        setIsChatsLoading(true);
        let loadedChats;
        try {
          loadedChats = await getChats(token, filters);
        } catch (e) {
          console.log(e);
          loadedChats = { count: 0, results: [] };
          setError(true);
        }
        setChats(loadedChats.results);
        setCount(loadedChats.count);
        setIsChatsLoading(false);
      };
      loadChats(user.token);
    }
  }, [filters, user.token]);

  const loadMoreMessages = async () => {
    let res;
    try {
      res = await getChats(user.token, { ...filters, page: activePage + 1 });
      setActivePage(activePage + 1);
    } catch (e) {
      setError(true);
      return;
    }

    setChats([...chats, ...res.results]);
  };

  const loadMessages = async (token) => {
    setActivePage(1);
    let res;
    try {
      res = await getChats(token, filters);
    } catch (e) {
      setError(true);
      return;
    }

    setChats(res.results);
  };

  const data = {
    chats,
    error,
    count,
    loadMoreMessages,
    isChatsLoading,
    loadMessages,
  };

  return <ChatContext.Provider value={data} {...props} />;
};
