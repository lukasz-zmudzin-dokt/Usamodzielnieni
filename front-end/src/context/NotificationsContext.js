import React, {
  useState,
  useContext,
  useEffect,
  useRef,
  createContext,
} from "react";
import { UserContext } from "context";
import proxy from "config/api";
import { paths } from "constants/paths";
import { userTypes } from "constants/userTypes";
import { compile } from "path-to-regexp";

const getNotifications = async (token, type, next) => {
  let url = proxy.notification + "list/all" + (next ? `?page=${next}` : "");
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };

  const response = await fetch(url, { method: "GET", headers });

  if (response.status !== 200) {
    throw response.status;
  }

  return response
    .json()
    .then((notifications) => mapNotifications(notifications, type));
};

const getNotificationsCount = async (token) => {
  let url = proxy.notification + "count/unread";
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };

  const response = await fetch(url, { method: "GET", headers });

  if (response.status !== 200) {
    throw response.status;
  }

  return (await response.json()).unread_count;
};

const markAsRead = async (token) => {
  let url = proxy.notification + "list/all/mark-as-read/";
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };

  const response = await fetch(url, { method: "POST", headers });

  if (response.status !== 200) {
    throw response.status;
  }

  return;
};

const deleteNotification = async (token, id) => {
  let url = proxy.notification + `delete/${id}/`;
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };

  const response = await fetch(url, { method: "DELETE", headers });

  if (response.status !== 200) {
    throw response.status;
  }

  return;
};

const deleteNotifications = async (token) => {
  let url = proxy.notification + "list/all/delete/";
  const headers = {
    Authorization: "Token " + token,
    "Content-Type": "application/json",
  };

  const response = await fetch(url, { method: "DELETE", headers });

  if (response.status !== 200) {
    throw response.status;
  }

  return;
};

const mapNotifications = (notifications, type) => ({
  notifications: notifications.results.map((notification) =>
    mapNotification(notification, type)
  ),
  next: notifications.next?.replace(/.*page=/, "") || undefined,
});

const mapNotification = (notification, type) => ({
  id: notification.slug,
  path: getPath(notification.app, notification.object_id, type),
  title: notification.text,
  time: new Date(notification.timestamp),
  unread: notification.unread,
});

const getPath = (appName, id, type) => {
  const isStandard = type === userTypes.STANDARD;
  if (appName.match(/^cv/)) {
    return isStandard
      ? compile(paths.MY_CVS)({})
      : compile(paths.CV_APPROVAL)({});
  } else if (appName.match(/^jobs/)) {
    return isStandard
      ? compile(paths.JOB_OFFERS)({})
      : compile(paths.MY_OFFERS)({});
  } else if (appName.match(/^account/)) {
    return compile(paths.USER)({});
  } else if (appName.match(/^blog/)) {
    return compile(paths.BLOG_POST)({ id });
  } else {
    return compile(paths.DASHBOARD)({});
  }
};

export const NotificationsContext = createContext(null);

export const NotificationsProvider = (props) => {
  const [notifications, setNotifications] = useState(null);
  const [count, setCount] = useState(null);
  const [next, setNext] = useState();
  const [error, setError] = useState(false);
  const socket = useRef();

  const user = useContext(UserContext);

  useEffect(() => {
    const loadNotifications = async (token, type) => {
      setError(false);
      let values;
      try {
        values = await Promise.all([
          getNotifications(token, type),
          getNotificationsCount(token),
        ]);
      } catch (e) {
        console.log(e);
        setError(true);
        return;
      }
      const [
        { notifications: loadedNotifications, next: loadedNext },
        loadedCount,
      ] = values;
      setCount(loadedCount);
      setNext(loadedNext);
      setNotifications(loadedNotifications);
    };
    if (user.token && user.type) {
      loadNotifications(user.token, user.type);
    }
  }, [user.token, user.type]);

  useEffect(() => {
    if (socket.current) {
      socket.current.close();
      socket.current = null;
    }
    if (user.token && user.type) {
      const url = proxy.wsNotification + "ws";
      try {
        socket.current = new WebSocket(url, user.token);
        // socket.current.onopen = (e) => console.log("onopen", e);
        socket.current.onmessage = (e) => {
          const newNotification = mapNotification(
            JSON.parse(e.data),
            user.type
          );
          setCount((prev) => prev + 1);
          setNotifications((prev) => [newNotification, ...prev]);
        };
        socket.current.onerror = (e) => {
          console.log(e);
          setError(true);
        };
      } catch (e) {
        console.log(e);
      }
    }
  }, [user.token, user.type]);

  const data = {
    notifications,
    count,
    error,
    next,
    socket,
    markAsRead: async () => {
      setCount(0);
      try {
        await markAsRead(user.token);
      } catch (e) {
        setError(true);
        return;
      }
    },
    deleteNotification: async (id) => {
      const notificationToRemove = notifications.find(
        (notification) => notification.id === id
      );
      if (notificationToRemove?.unread && count > 0) {
        setCount((prev) => prev - 1);
      }
      setNotifications((prev) =>
        prev.filter((notification) => notification.id !== id)
      );
      try {
        await deleteNotification(user.token, id);
      } catch (e) {
        setError(true);
        return;
      }
    },
    deleteNotifications: async () => {
      setCount(0);
      setNotifications([]);
      try {
        await deleteNotifications(user.token);
      } catch (e) {
        setError(true);
        return;
      }
    },
    loadMoreNotifications: async () => {
      setNext(null);
      let res;
      try {
        res = await getNotifications(user.token, user.type, next);
      } catch (e) {
        setError(true);
        return;
      }
      setNext(res.next);
      setNotifications((prev) => [...prev, ...res.notifications]);
    },
  };
  return <NotificationsContext.Provider value={data} {...props} />;
};
