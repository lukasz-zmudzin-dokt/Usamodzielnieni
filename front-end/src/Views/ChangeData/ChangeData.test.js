import React from "react";
import {
  render,
  waitForElement,
  fireEvent,
  wait,
} from "@testing-library/react";
import ChangeData from "./ChangeData";
import { createMemoryHistory } from "history";
import { UserContext } from "context";
import { Router } from "react-router-dom";

// jest.mock("./components", () => ({
//   CompanyForm: () => <div>Company</div>,
//   FacilityForm: () => <div>Facility</div>,
// }));

const renderWithRouter = (
  ui,
  {
    route = "/changeData/abc",
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  let context = {
    token: "abc",
  };
  return {
    ...render(
      <UserContext.Provider value={context}>
        <Router history={history}>{ui}</Router>
      </UserContext.Provider>
    ),
    history,
  };
};

describe("ChangeData", () => {
  let failFetch;
  let data = {
    first_name: "user",
    last_name: "user",
    phone_number: "+48123123123",
    facility_name: "abc",
    facility_address: {
      city: "new",
      street: "city",
      street_number: "12",
      postal_code: "12-123",
    },
  };
  beforeAll(() => {
    global.fetch = jest.fn().mockImplementation((input, init) => {
      return new Promise((resolve, reject) => {
        if (failFetch) {
          resolve({ status: 500 });
        }
        switch (init.method) {
          case "PUT":
            resolve({ status: 200 });
            break;
          case "GET":
            resolve({ status: 200, json: () => Promise.resolve(data) });
            break;
          default:
            reject({});
            break;
        }
      });
    });
  });

  beforeEach(() => {
    data = {
      first_name: "user",
      last_name: "user",
      phone_number: "+48123123123",
      facility_name: "abc",
      facility_address: {
        city: "new",
        street: "city",
        street_number: "12",
        postal_code: "12-123",
      },
    };
  });

  it("should match snapshot", async () => {
    const { container, getByText } = renderWithRouter(<ChangeData />);

    await waitForElement(() => getByText("Adres placówki"));

    expect(container).toMatchSnapshot();
  });

  it("should redirect after sending data(facility form)", async () => {
    const { getByText, history, getByLabelText } = renderWithRouter(
      <ChangeData />
    );

    await waitForElement(() => getByText("Adres placówki"));

    fireEvent.change(getByLabelText("Imię"), {
      target: { value: "abc" },
    });
    fireEvent.change(getByLabelText("Nazwisko"), {
      target: { value: "abc" },
    });
    fireEvent.change(getByLabelText("Numer telefonu"), {
      target: { value: "adsasd" },
    });
    fireEvent.change(getByLabelText("Nazwa placówki"), {
      target: { value: "asdad" },
    });
    fireEvent.change(getByLabelText("Ulica"), {
      target: { value: "ges" },
    });
    fireEvent.change(getByLabelText("Numer budynku"), {
      target: { value: "cde" },
    });
    fireEvent.change(getByLabelText("Nazwa miasta"), {
      target: { value: "abc" },
    });
    fireEvent.change(getByLabelText("Kod pocztowy"), {
      target: { value: "21-115" },
    });
    fireEvent.click(getByText("Prześlij zmiany"));

    await wait(() => {
      expect(fetch).toHaveBeenCalled();
    });

    expect(history.location.pathname).toEqual("/userList");
  });

  it("should redirect after sending data(company form)", async () => {
    data = {
      first_name: "user",
      last_name: "user",
      phone_number: "+48123123123",
      company_name: "abc",
      company_address: {
        city: "new",
        street: "city",
        street_number: "12",
        postal_code: "12-123",
      },
      nip: "123123123",
    };
    const { getByText, history, getByLabelText } = renderWithRouter(
      <ChangeData />
    );

    await waitForElement(() => getByText("Adres firmy"));

    fireEvent.change(getByLabelText("Imię"), {
      target: { value: "abc" },
    });
    fireEvent.change(getByLabelText("Nazwisko"), {
      target: { value: "abc" },
    });
    fireEvent.change(getByLabelText("Numer telefonu"), {
      target: { value: "adsasd" },
    });
    fireEvent.change(getByLabelText("Nazwa firmy"), {
      target: { value: "asdad" },
    });
    fireEvent.change(getByLabelText("Ulica"), {
      target: { value: "ges" },
    });
    fireEvent.change(getByLabelText("Numer budynku"), {
      target: { value: "cde" },
    });
    fireEvent.change(getByLabelText("Nazwa miasta"), {
      target: { value: "abc" },
    });
    fireEvent.change(getByLabelText("Kod pocztowy"), {
      target: { value: "21-115" },
    });
    fireEvent.click(getByText("Prześlij zmiany"));

    await wait(() => {
      expect(fetch).toHaveBeenCalled();
    });

    expect(history.location.pathname).toEqual("/userList");
  });

  it("should redirect after click back button", async () => {
    const { getByText, history } = renderWithRouter(<ChangeData />);

    await waitForElement(() => getByText("Adres placówki"));

    fireEvent.click(getByText("<"));

    expect(history.location.pathname).toEqual("/userList");
  });

  it("should render CompanyForm when nip is in data", async () => {
    data = {
      first_name: "user",
      last_name: "user",
      phone_number: "+48123123123",
      company_name: "abc",
      company_address: {
        city: "new",
        street: "city",
        street_number: "12",
        postal_code: "12-123",
      },
      nip: "123123123",
    };
    const { getByText } = renderWithRouter(<ChangeData />);

    await waitForElement(() => getByText("Adres firmy"));
  });

  it("should render fail message if api fails", async () => {
    failFetch = true;
    const { getByText } = renderWithRouter(<ChangeData />);

    await waitForElement(() =>
      getByText("Nie udało się pobrać danych użytkownika.")
    );
  });
});
