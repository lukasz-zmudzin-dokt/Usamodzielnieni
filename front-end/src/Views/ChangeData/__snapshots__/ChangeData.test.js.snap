// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeData should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card"
    >
      <h2
        class="changeData__header card-header"
      >
        <button
          class="changeData__back btn btn-outline-warning"
          type="button"
        >
          &lt;
        </button>
        Popraw dane użytkownika:
         
      </h2>
      <div
        class="card-body"
      >
        <form
          class="changeData__form"
        >
          <div
            class="changeData__wrapper"
          >
            <div
              class="changeData__wrapper__card card bg-light"
            >
              <div
                class="card-header"
              >
                Dane osobowe
              </div>
              <div
                class="card-body"
              >
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="firstName"
                  >
                    Imię
                  </label>
                  <input
                    class="form-control"
                    data-testid="default"
                    id="firstName"
                    maxlength="30"
                    minlength="1"
                    placeholder="Imię"
                    type="text"
                    value="user"
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="lastName"
                  >
                    Nazwisko
                  </label>
                  <input
                    class="form-control"
                    data-testid="default"
                    id="lastName"
                    maxlength="30"
                    minlength="1"
                    placeholder="Nazwisko"
                    type="text"
                    value="user"
                  />
                </div>
                <div
                  class="form-group"
                  invalid="Podaj numer telefonu w formacie: +48123123123"
                >
                  <label
                    class="form-label"
                    for="phoneNumber"
                  >
                    Numer telefonu
                  </label>
                  <input
                    class="form-control"
                    id="phoneNumber"
                    type="tel"
                    value="+48123123123"
                  />
                </div>
              </div>
            </div>
            <div
              class="changeData__wrapper__card card bg-light"
            >
              <div
                class="card-header"
              >
                Adres placówki
              </div>
              <div
                class="card-body"
              >
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="facilityName"
                  >
                    Nazwa placówki
                  </label>
                  <input
                    class="form-control"
                    data-testid="default"
                    id="facilityName"
                    maxlength="100"
                    minlength="1"
                    placeholder="Nazwa placówki"
                    type="text"
                    value="abc"
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="street"
                  >
                    Ulica
                  </label>
                  <input
                    class="form-control"
                    data-testid="default"
                    id="street"
                    maxlength="120"
                    minlength="1"
                    placeholder="Ulica"
                    type="text"
                    value="city"
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="streetNumber"
                  >
                    Numer budynku
                  </label>
                  <input
                    class="form-control"
                    data-testid="default"
                    id="streetNumber"
                    maxlength="20"
                    minlength="1"
                    placeholder="Numer budynku"
                    type="text"
                    value="12"
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="city"
                  >
                    Nazwa miasta
                  </label>
                  <input
                    class="form-control"
                    data-testid="default"
                    id="city"
                    maxlength="40"
                    minlength="1"
                    placeholder="Nazwa miasta"
                    type="text"
                    value="new"
                  />
                </div>
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="postalCode"
                  >
                    Kod pocztowy
                  </label>
                  <input
                    class="form-control"
                    data-testid="default"
                    id="postalCode"
                    maxlength="6"
                    minlength="1"
                    pattern="[0-9]{2}[-][0-9]{3}"
                    placeholder="Kod pocztowy"
                    type="text"
                    value="12-123"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ml-0 mr-0 mt-3 justify-content-center row"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Prześlij zmiany
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
